pkg/operator/ceph/cluster/nodedaemon/ceph-exporter.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodedaemon
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"strconv"
+
+	"github.com/pkg/errors"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/config"
+	"github.com/rook/rook/pkg/operator/ceph/controller"
+
+	cephver "github.com/rook/rook/pkg/operator/ceph/version"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	monitoringPath               = "/etc/ceph-monitoring/"
+	serviceMonitorFile           = "ceph-exporter-service-monitor.yaml"
+	sockDir                      = "/run/ceph"
+	perfCountersPrioLimit        = "5"
+	statsPeriod                  = "5"
+	DefaultMetricsPort    uint16 = 9926
+)
+
+// createOrUpdateCephExporter is a wrapper around controllerutil.CreateOrUpdate
+func (r *ReconcileNode) createOrUpdateCephExporter(node corev1.Node, tolerations []corev1.Toleration, cephCluster cephv1.CephCluster, cephVersion *cephver.CephVersion) (controllerutil.OperationResult, error) {
```

Comment:
Some of the integration tests are failing due to running with an older version of Ceph. The failing tests are running with Pacific. For example, the helm suite is failing because the exporter daemon tries to start, but shows this error in the pod description:
```
      Message:      failed to create shim: OCI runtime create failed: runc create failed: unable to start container process: exec: "/usr/bin/ceph-exporter": stat /usr/bin/ceph-exporter: no such file or directory: unknown
```
So we need to add a condition that will only start the exporter on the expected versions of Ceph. Which versions of Ceph exactly contain the exporter? 17.2.5 or newer, right? So we will need to add a check such as the following to skip reconciling the exporter:
```go
if !cephVersion.IsAtLeast(cephver.CephVersion{Major: 17, Minor: 2, Extra: 5}) {
    logger.Infof("Skipping exporter reconcile on ceph version %q", cephver.String())
    return controllerutil.OperationResultNone, nil
}
```